src/LayerContext.js

```diff
@@ -0,0 +1,11 @@
+import React, { useContext } from 'react'
+
+const LayerContext = React.createContext(0)
+const LayerProvider = LayerContext.Provider
+
+const useLayerBasedZindex = (propZIndex = 0, defaultZIndex = 0) => {
+    const contextZIndex = useContext(LayerContext) || 0
```

Comment:
I don't think you need the `|| 0` as since if there is no Provider defined up the tree, the context will default to `0` as defined in `React.createContext(0)`.

Reply:
Maybe it would.... In v16.3 I tried this and believe it returned `undefined` if there was no provider up the tree.... I'll give it a try.

Reply:
You were correct. Fixed in https://github.com/dhis2/ui-core/pull/582/commits/e3cb382a80a13951940a28027329be9ae25df869